### ISSUE TITLE
temporary failure causes "publish" to get into a borked state

### ISSUE THREAD

**Issue by MonsieurNicolas** [stellar/stellar-core#1763]
# Issue type
  bug

# Version found
9.2.0

# Your environment and setup
Ubuntu 14.04 LTS

# Issue description
Box ran out of memory for a few seconds, triggering a bunch of failures during publish.

## Actual result
Some steps from publish failed (gzip), yet core proceeded in trying to upload files, and of course, kept failing trying to upload (it looks like it was retrying to upload a non existent file forever).

Resolution was to restart core to force it to retry to create the zip files.

## Expected result
Failures of any step should cause that step to be retried instead of continuing the workflow.

## Supporting files

```
[History INFO] Publishing 1 queued checkpoints [19594943-19594943]: Awaiting 0/0 prerequisites of: publish-012afebf
[Ledger INFO] Closed ledger: [seq=19594943, hash=7ab892]
[Process ERROR] posix_spawn() failed: Cannot allocate memory [ProcessManagerImpl.cpp:432]
[Process ERROR] Error starting process: posix_spawn() failed [ProcessManagerImpl.cpp:489]
[Process ERROR] When running: gzip -c /data/buckets/tmp/snapshot-ec6077c85b24ec82/scp-012afebf.xdr [ProcessManagerImpl.cpp:490]
```
...later...
```
The user-provided path /data/buckets/tmp/snapshot-ec6077c85b24ec82/scp-012
[Process WARNING] process 3440 exited 255: aws s3 cp --region eu-west-1 /data/buckets/tmp/snapshot-ec6077c85b24ec82/scp-012afebf.xdr.gz s3://history.stellar.org/prd/core-live/core_live_002/scp/01/2a/fe/scp-012afebf.xdr.gz
```

**Comment by marta-lokhova** [comment]
This looks to me exactly like https://github.com/stellar/stellar-core/issues/1577, which appears to be fixed in 10.0.0. I wrote a small test re-producing this scenario (same work tree, with process failing to start), and running it with the new code did not produce this problem.

**Comment by MonsieurNicolas** [comment]
ah yes you are probably right
